app/domain/authentication/authn_jwt/validate_and_decode/validate_and_decode_token.rb

```diff
@@ -31,12 +43,23 @@ def call
 
         private
 
+        # Don't do memoization here because otherwise interface won't change between different requests
+        def fetch_signing_key_interface
```

Comment:
i would call this `signing_key_interface` and just call this function when you need it instead of explicitly call it from the `call` function`